coordinator/src/service.rs

```diff
@@ -315,24 +366,21 @@ impl CoordService {
             let _ = sender.send(Err(e)).await;
             return;
         }
+        let executor = QueryExecutor::new(option, kv_inst, meta, sender.clone(), metrics);
```

Comment:
could use `QueryExecutorFactory` instead, so you don't need to pass in `metrics: Arc<MetricsRegister>` here